subsys/bluetooth/host/gatt.c

```diff
@@ -4761,6 +4761,15 @@ static void parse_read_by_uuid(struct bt_conn *conn,
 		uint16_t handle;
 		uint16_t len;
 
+		len = MIN(rsp->len, length);
+		if (len < sizeof(struct bt_att_data)) {
+			LOG_WRN("Bad peer: ATT read-by-uuid rsp: invalid ATTR data len %u", len);
+			params->func(conn, BT_ATT_ERR_UNLIKELY, params, NULL, 0);
```

Comment:
Would
```suggestion
			params->func(conn, BT_ATT_ERR_INVALID_ATTRIBUTE_LEN, params, NULL, 0);
```

Make more sense? We should also document what ATT error we provide to the upper layers if this happens